interfaces/C/ssids.f90

```diff
@@ -225,6 +232,90 @@ subroutine spral_ssids_analyse(ccheck, n, corder, cptr, crow, cval, cakeep, &
   call copy_inform_out(finform, cinform)
 end subroutine spral_ssids_analyse
 
+subroutine spral_ssids_analyse_topology(ccheck, n, order, ptr, row, val, cakeep, &
+   coptions, cinform, nregions, cregions) bind(C)
+use spral_ssids_ciface
+implicit none
+
+logical(C_BOOL), intent(in), value :: ccheck
+integer(C_INT), intent(in), value :: n, nregions
+integer(C_INT), intent(inout), dimension(n), optional :: order
+integer(C_INT64_T), intent(in), dimension(n+1) :: ptr
+type(C_PTR), intent(inout) :: cakeep
+type(spral_ssids_options), intent(in) :: coptions
+type(spral_ssids_inform), intent(out) :: cinform
+integer(C_INT), intent(in), dimension(ptr(n+1)-coptions%array_base) :: row
+real(C_DOUBLE), dimension(ptr(n+1)-coptions%array_base), optional :: val
+type(spral_numa_region), intent(in), dimension(nregions) :: cregions
+
+integer(C_INT64_T), dimension(:), allocatable :: ptr_find
+integer(C_INT), dimension(:), allocatable :: row_find, order_find
+logical :: fcheck
+type(ssids_akeep), pointer :: fakeep
+type(ssids_options) :: foptions
+type(ssids_inform) :: finform
+type(numa_region), dimension(:), allocatable :: fregions
+integer(C_INT), dimension(:), pointer :: fgpus
+
+logical :: cindexed
+integer :: i
+
+! Copy options in first to find out whether we use Fortran or C indexing
+call copy_options_in(coptions, foptions, cindexed)
+
+! Translate topology
+allocate(fregions(nregions))
+do i = 1, nregions
+   fregions(i)%nproc = cregions(i)%nproc
+   if (cregions(i)%ngpu .gt. 0) then
+      allocate(fregions(i)%gpus(cregions(i)%ngpu))
+      call C_F_POINTER(cregions(i)%gpus, fgpus, shape=(/ cregions(i)%ngpu /))
+      fregions(i)%gpus(:) = fgpus
+   else ! no gpus, need to allocate zero array
+      allocate(fregions(i)%gpus(0))
+   end if
+end do
+
+! Translate arguments
+fcheck = ccheck
+if (cindexed) then
+   allocate(ptr_find(n+1))
+   ptr_find = ptr + 1
+   allocate(row_find(ptr(n+1)-1))
```

Comment:
I think this has to be either `allocate(row_find(ptr(n+1)))` or `allocate(row_find(ptr_find(n+1)-1))`

Reply:
Good catch, thank you!